src/SubscribeBlockTracker.test.ts

```diff
@@ -142,6 +142,21 @@ describe('SubscribeBlockTracker', () => {
         });
       });
 
+      it('should return the same promise if called multiple times', async () => {
+        recordCallsToSetTimeout();
+
+        await withSubscribeBlockTracker(async ({ blockTracker }) => {
+          const promiseToGetLatestBlock1 =
+            blockTracker[methodToGetLatestBlock]();
+          const promiseToGetLatestBlock2 =
+            blockTracker[methodToGetLatestBlock]();
+
+          expect(promiseToGetLatestBlock1).toStrictEqual(
+            promiseToGetLatestBlock2,
+          );
```

Comment:
Similar as above, should this use `toBe`?

``` suggestion
          expect(promiseToGetLatestBlock1).toBe(promiseToGetLatestBlock2);

Reply:
I tried that initially but then the test case started failing: 

```
expect(received).toBe(expected) // Object.is equality

    If it should pass with deep equality, replace "toBe" with "toStrictEqual"

    Expected: {}
    Received: serializes to the same string
```

We indeed would want to use `toBe` since it would test that the promise references are indeed pointing to the same object. Do you know how we could get around this error?

Reply:
Oh I see, because we're returning the promise representing the previous pending request, right.

I'm not sure that they would indeed be the same, depending on how Promise unwrapping works for asynchronous functions.

Consider that when an asynchronous function is called, a Promise is created representing the return value. That's why an asynchronous function can return a primitive value, and the caller receives it as `Promise<...>`.

There is a thing called "Promise unwrapping" (I think?) where a Promise returned directly from an asynchronous function is "unwrapped" so that we don't get a silly double-promise. Like that's what's happening here - an equivalent Promise is getting returned, but the inner Promise is unwrapped leaving just the outer Promise, which is distinct to each call.

Reply:
Yeah that was my thinking since we returned the promise stored in the class variable. 

Thanks a lot for this explanation! In this case, we can probably remove these tests altogether, since it should be already covered by https://github.com/MetaMask/eth-block-tracker/pull/284/files#diff-59b2763dcfc8532df91ccd2df5bfd3051d465e6eef759260d2430f4921d28c75R246

Reply:
Yes I think you're right, those tests cover this